doc/man3/OPENSSL_LH_COMPFUNC.pod

```diff
@@ -162,34 +179,48 @@ that is provided by the caller):
 B<lh_I<TYPE>_error>() can be used to determine if an error occurred in the last
 operation.
 
+OPENSSL_LH_new() is the same as the lh_TYPE_new() except that it is not type
+specific. So instead of returning an LHASH_OF(TYPE) value it returns a void *.
+In the same way the functions OPENSSL_LH_free, OPENSSL_LH_flush,
+OPENSSL_LH_insert, OPENSSL_LH_delete, OPENSSL_LH_retrieve, OPENSSL_LH_doall,
+OPENSSL_LH_doall_arg, and OPENSSL_LH_error are equivalent to the similarly named
+lh_TYPE functions except that they return or use a void * where the equivalent
+lh_TYPE function returns or uses a TYPE * or LHASH_OF(TYPE) *. lh_TYPE functions
+are implemented as type checked wrappers around the OPENSSL_LH functions. Most
+applications should not call the OPENSSL_LH functions directly.
```

Comment:
man-pages(7) doesn't say anything specific about how types should be formatted.  Looking through diverse man pages, I see them sometimes in bold, and sometimes in italics.  In our pages, we tend to use bold for the verbatim types.  What we've also done lately, when there's that metaphorical "TYPE", is to make it bold italics (i.e. italics markup inside the bold markup).

So in this paragraph, I can see the following markup additions:

`void *` => `B<void *>`
`LHASH_OF(TYPE)` => `B<LHASH_OF(I<TYPE>)>`
`LHASH_OF(TYPE) *` => `B<LHASH_OF(I<TYPE>) *>`
`TYPE *` => `B<I<TYPE> *>`

Functions need some markup too.  If we have `()` at the end of the function name, perldoc does that for us.  I see that you haven't done that for `OPENSSL_LH_free`, `OPENSSL_LH_flush`, ...
There are times when perldoc needs a bit of help, though, so please change:

`lh_TYPE_new()` => `B<lh_I<TYPE>_new>()`
`lh_TYPE functions` => `B<lh_I<TYPE>> functions`
`OPENSSL_LH functions` => `B<OPENSSL_LH> functions`